libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -697,10 +689,21 @@ PropertyValue YamlConfiguration::getValidatedProcessorPropertyForDefaultTypeInfo
     if (defaultType == typeid(int64_t)) {
       coercedValue = propertyValueNode.as<int64_t>();
     } else if (defaultType == typeid(uint64_t)) {
-      try {
-        coercedValue = propertyValueNode.as<uint64_t>();
-      } catch (...) {
-        coercedValue = propertyValueNode.as<std::string>();
+      const auto uValue = propertyValueNode.as<uint64_t>(0);
+
+      // parsing uint64_t may have failed
+      if (uValue == 0) {
+        const auto sValue = propertyValueNode.as<std::string>();
+
+        // parsing uint64_t did not fail, the node was a 0
+        if (sValue == "0") {
+          coercedValue = uValue;
+        } else {
+          // parsing uint64_t really failed
+          coercedValue = sValue;
+        }
+      } else {
+        coercedValue = uValue;
```

Comment:
I agree that using exceptions for flow control is not nice, but the new code is so much more complicated than the old code that I would prefer to keep the old version.

Reply:
This is a good alternative, thanks.

Reply:
This is a good alternative, thanks.